### ISSUE TITLE
Sharing OTP code using password is not working

### ISSUE THREAD

**Issue by emaddaou** [matteo-convertino/otpmanager-nextcloud#49]
Thank you again for the great Plugin; 

I wanted to share an OTP with family member, when asked to set a password I did and shared the password with my family member, however, when they use it to unlock the OTP code, it keeps saying that the password is not valid or it's not unlocking the OTP code, no matter how many times and what type of password I choose, still my family member can't unlock the OTP code; 

Is it possible to make the password an optional choice here? We really don't need password to lock and unlock the code since my family member already using their password to unlock the OTP vault; 

I look forward for your response at your earliest convenience;

**Comment by matteo-convertino** [comment]
I'm sorry if I answer only now. The sharing password is required to allow the functionality to work.
It doesn't seem to have ever happened as a problem, do you see any errors in the nextcloud logs?

> "or it's not unlocking the OTP code?"

What do you mean by this? There should only be one error, namely that the password is wrong.

Just to be sure, as silly as it may seem, the process should go something like this:
- when sharing enter any password, even "aa"
- the other user will have to enter "aa" to unlock the account
- if you don't remember the password used to share, you can simply cancel the sharing and do it again with a new password

**Comment by hdwverden** [comment]
I'm running in the same issue. It gets stuck trying to unlock the shared OTP.

The log shows the following error:
OCA\OtpManager\Controller\SharedAccountApiController::unlock(): Argument #1 ($accountId) must be of type int, null given, called in [cloud URL]/lib/private/AppFramework/Http/Dispatcher.php on line 232 in file '[cloud URL]/apps/otpmanager/lib/Controller/SharedAccountApiController.php' line 42

**Comment by Liam-Weitzel** [comment]
Having the same issue

![screenshot](https://github.com/user-attachments/assets/d75553d8-c4c7-4638-92b4-fa6f2638897a)

**Comment by Liam-Weitzel** [comment]
Works for me now :)

**Comment by sjlongland** [comment]
> Works for me now :)

Mind sharing what happened between the 11th and 16th of November that made it work?  It's been not working for me for quite some time now.

**Comment by emaddaou** [comment]
Thank you

**Comment by Liam-Weitzel** [comment]
> > Works for me now :)
> 
> Mind sharing what happened between the 11th and 16th of November that made it work? It's been not working for me for quite some time now.

I used the OTP manager android app to accept the sharing request and it worked magically. No idea if this is what resolved the issue.

**Comment by sjlongland** [comment]
Right, so it's "fixed" (or perhaps never broken) in the Android application… but for those of us that use a real computer, it's still broken.

So sorry, not completed, can we please look into this?

**Comment by Liam-Weitzel** [comment]
Correct, I'm still getting the same error on the web app (hadn't checked until now). @emaddaou were you able to resolve this issue?

**Comment by sjlongland** [comment]
@emaddaou Can we please look into fixing this?  Issue #64 clearly shows the problem still exists and I can confirm the issue here as well.

**Comment by tccadam** [comment]
Yup, I just attempted to share with 3 people and spent an hour trying to figure it out and eventually found my way to this forum.  I hope someone can fix it and update the app, because I love it and with the ability to share these it would become even more valuable.

**Comment by tccadam** [comment]
HEY! I fixed it!

Here is how, hopefully @matteo-convertino can use this in the next update: 

--Edit this file in your nextcloud installation folder:

apps/otpmanager/js/otpmanager-main.js

--find this:

accountId:n.account_id,

--CHANGE IT TO:

accountId:n.id,

--THAT FIXED IT! (you may have to reload the browser in incognito mode and / or hard reload it using Ctrl+Shift+R in Chrome)

--NOW, to fix the SOURCE FILE, which should fix it if the app gets reloaded or updated I hope, edit this file:

apps/otpmanager/src/js/modals/UnlockSharedAccount.js

--Find the line:

accountId: sharedAccountToUnlock.account_id,

--CHANGE IT TO:

accountId: sharedAccountToUnlock.id, 

--NOW, the section looks like this:

    axios
      .post(generateOcsUrl("/apps/otpmanager/share/unlock"), {
        accountId: sharedAccountToUnlock.id,
        currentPassword: secret.passwordHash,
        tempPassword: values.tempPassword,
      })

--99.9% of the credit goes to ChatGPT which is the greatest thing man has ever invented and .1% credit goes to my OCD.

Thanks @matteo-convertino for the great nextcloud addon, I hope you keep updating it, I love it, all my codes go in there now.

Adam

**Comment by Liam-Weitzel** [comment]
> HEY! I fixed it!

@tccadam Amazing!! Thank you for taking the time to look into this. Will you be raising a PR for this fix? Otherwise I'm happy to do it, just wanted to make sure I'm not stealing your credit :)

**Comment by tccadam** [comment]
Thanks @Liam-Weitzel !  I don't even know what a PR is, so yes, please go ahead and do that.

**Comment by sjlongland** [comment]
For the record: latest NextCloud (v32) and OTP Manager (v0.5.4) still sits there doing this:

<img width="1236" height="629" alt="Image" src="https://github.com/user-attachments/assets/e08af30b-7bf1-44e2-b32c-8f91fe3b09cc" />

It's been sitting there in my browser like that for the past 20 minutes or so.  If that's "fixed", I'd hate to see what broken looked like.

**Comment by Liam-Weitzel** [comment]
@sjlongland the [PR I opened](https://github.com/matteo-convertino/otpmanager-nextcloud/pull/70) with changes suggested by @tccadam is still not merged as the maintainer seems to be inactive. @tccadam shared his fix which you can apply locally using his guide as I did.

**Comment by sjlongland** [comment]
> [@sjlongland](https://github.com/sjlongland) the [PR I opened](https://github.com/matteo-convertino/otpmanager-nextcloud/pull/70) with changes suggested by [@tccadam](https://github.com/tccadam) is still not merged as the maintainer seems to be inactive. [@tccadam](https://github.com/tccadam) shared his fix which you can apply locally using his guide as I did.

I'm aware of that… I'm trying to poke the maintainer into actually reviewing those changes and getting the problem fixed.  At the very least, the issue should be re-opened.  I cannot do that as I don't have permission to do so.

**Comment by matteo-convertino** [comment]
Guys, I'm alive! Jokes aside, I'm really happy there are still people active on this project who want the app to continue receiving updates.

To be honest, I haven't maintained the app in the last two years because:
- it did what it was supposed to do without too many problems (maybe 🫣 sorry for this issue)
- it was a project I did in my spare time, and that spare time was starting to run out
- the time I invested in this project was starting to become too much compared to the donations I could have received (almost zero). I sincerely thank the few people who did. It's mainly thanks to you that I've decided to release some more updates soon.

That said, I've been working on this project again for a few months now (as you can see from the new "refactoring" branch). I've done and am finishing a fairly extensive code restructuring and update. For the end user, it won't change much for now, but it's significantly helped me (or anyone else who wants to contribute) in development.

I'm currently working on the mobile app and as soon as I'm done there and tested everything, I'll be able to release v2 of OTP Manager.

I hope you're happy with this and I hope to publish the release as soon as possible 🚀.
In the meantime, I'll reopen the issue.

**Comment by sjlongland** [comment]
Ahh okay, no problems.  Yeah, real life often means projects take a back seat, and I'm fine with that.

It was the radio silence and the incorrect bug status that was irritating me the most.  I was worried the project had become abandonware and I'd have to fork it myself.  At least re-opening the issue means there's a reminder that the issue still exists, and can be dealt with once the re-factoring work is complete. :-)

**Comment by matteo-convertino** [comment]
I am closing this issue as this bug should now be fixed in the [new release](https://github.com/matteo-convertino/otpmanager-nextcloud/releases/tag/v1.0.1) 🚀.

Please do not hesitate to reopen the issue if this problem persists ❤️.

**Comment by sjlongland** [comment]
Can confirm, it is working well now.  I'm finally able to unlock the secret that's been kicking around unreadable for the past 6 months. :-)

<img width="1073" height="674" alt="Image" src="https://github.com/user-attachments/assets/42a70e9e-3673-45f6-8990-00448403a266" />
